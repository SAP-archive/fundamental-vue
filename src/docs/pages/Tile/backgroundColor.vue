<title>Tile with background color accent</title>

<template>
  <FdTile title="Title" description="Description" backgroundColor="accent-2" />
</template>
