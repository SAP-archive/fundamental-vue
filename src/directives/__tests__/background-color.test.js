import { mount, createLocalVue } from "@vue/test-utils";
import backgroundColor from "./../background-color";

describe("background-color directive", () => {
  it("attaches correct color class when used with arg", () => {
    const localVue = createLocalVue();
    const wrapper = mount(
      {
        directives: { backgroundColor },
        template: `<div v-background-color:red />`
      },
      { localVue }
    );
    expect(wrapper.element).toMatchSnapshot();
  });

  it("attaches correct color class when used with value", async () => {
    const localVue = createLocalVue();
    const wrapper = mount(
      {
        data() {
          return { color: "red" };
        },
        directives: { backgroundColor },
        template: `<div v-background-color="color" />`
      },
      { localVue }
    );
    expect(wrapper.element).toMatchSnapshot();
    wrapper.setData({ color: "green" });
    await localVue.nextTick();

    expect(wrapper.element).toMatchSnapshot();
  });
});
