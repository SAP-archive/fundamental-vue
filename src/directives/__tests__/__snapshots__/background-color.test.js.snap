// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`background-color directive attaches correct color class when used with arg 1`] = `
<div
  class="fd-has-background-color-red"
/>
`;

exports[`background-color directive attaches correct color class when used with value 1`] = `
<div
  class="fd-has-background-color-red"
/>
`;

exports[`background-color directive attaches correct color class when used with value 2`] = `
<div
  class="fd-has-background-color-green"
/>
`;
